examples/simple-genai-server/main.go

```diff
@@ -0,0 +1,240 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"agones.dev/agones/pkg/util/signals"
+	sdk "agones.dev/agones/sdks/go"
+)
+
+// Main starts a server that serves as an example of how to integrate GenAI endpoints into your dedicated game server.
+func main() {
+	sigCtx, _ := signals.NewSigKillContext()
+
+	port := flag.String("port", "7654", "The port to listen to traffic on")
+	genAiEndpoint := flag.String("genAiEndpoint", "", "The full base URL to send API requests to to simulate computer (NPC) responses to user input")
+	genAiContext := flag.String("GenAiContext", "", "Context for the GenAI endpoint")
+	prompt := flag.String("Prompt", "", "The first prompt for the GenAI endpoint")
+	simEndpoint := flag.String("simEndpoint", "", "The full base URL to send API requests to to simulate user input")
+	simContext := flag.String("SimContext", "", "Context for the Sim endpoint")
+	numChats := flag.Int("NumChats", 1, "Number of back and forth chats between the sim and genAI")
+
+	var simConn *connection
+	var compConn * connection
+
+	flag.Parse()
+	if ep := os.Getenv("PORT"); ep != "" {
+		port = &ep
+	}
+	if sc := os.Getenv("SimContext"); sc != "" {
+		simContext = &sc
+	}
+	if gac := os.Getenv("GenAiContext"); gac != "" {
+		genAiContext = &gac
+	}
+	if p := os.Getenv("Prompt"); p != "" {
+		prompt = &p
+	}
+	if se := os.Getenv("SimEndpoint"); se != "" {
+		simEndpoint = &se
+		log.Printf("Creating Client at endpoint %s", *simEndpoint)
+		simConn = initClient(*simEndpoint, *simContext)
+	}
+	if gae := os.Getenv("GenAiEndpoint"); gae != "" {
+		genAiEndpoint = &gae
+		log.Printf("Creating Client at endpoint %s", *genAiEndpoint)
+		compConn = initClient(*genAiEndpoint, *genAiContext)
+	}
+	if nc := os.Getenv("NumChats"); nc != "" {
+		num, err := strconv.Atoi(nc)
+		if err != nil {
+			log.Fatalf("Could not parse NumChats: %v", err)
+		}
+		numChats = &num
+	}
+
+	log.Print("Creating SDK instance")
+	s, err := sdk.NewSDK()
+	if err != nil {
+		log.Fatalf("Could not connect to sdk: %v", err)
+	}
+
+	log.Print("Starting Health Ping")
+	// TODO: Is "cancel" still being used even if no method explicitly uses it?
+	ctx, _ := signals.NewSigKillContext()
+	go doHealth(s, ctx)
+
+	log.Print("Marking this server as ready")
+	if err := s.Ready(); err != nil {
+		log.Fatalf("Could not send ready message")
+	}
+
+	// Start up TCP listener so the user can interact with the GenAI endpoint manually
+	if simConn == nil {
+		go tcpListener(port, s, compConn)
```

Comment:
This is selecting between "interactive" mode and "autonomous" mode. Can the function names / comment make that clearer?